### PR TITLE
refactor(escrow): replace withdrawal SMT with an on-chain nonce bitmap

### PR DESCRIPTION
**Pull request by sebatustra**
Replace withdrawal SMT with an on-chain nonce bitmap

Draft — escrow program only. The indexer/operator is not migrated yet, so the workspace  does not build as a pair. See Not in this PR.

What

Withdrawal replay protection moves from a Sparse Merkle Tree to a fixed 8 KB bitmap PDA holding one bit per nonce. The nonce model is unchanged: generation = nonce / 65536, bit = nonce % 65536, with one account reused across generations.

Breaking interface changes

  - ResetSmtRoot → RotateBitmap. Discriminator stays 8; the arg becomes expected_generation.
  - New WithdrawalBitmap account, seeds ["withdrawal_bitmap", instance], discriminator 3. Created by CreateInstance, so every instance has one by construction.
  - Instance drops withdrawal_transactions_root and current_tree_index (107 → 67 bytes).
  - ReleaseFunds drops new_withdrawal_root and sibling_proofs; instance becomes read-only, since it only signs the transfer now.
  - CreateInstance gains a withdrawal_bitmap account and a bitmap_bump arg.
  - Custom errors 11–14 renumbered: InvalidWithdrawalBitmap, NonceAlreadyUsed, NonceOutsideCurrentGeneration, UnexpectedGeneration.
  - ReleaseFundsEvent loses new_withdrawal_root (177 → 145 bytes); RotateBitmapEvent gains new_generation.

  Security

  The at-most-once property is preserved by two composing mechanisms: within a generation the bit blocks replay, and since generation only increments, a nonce from a superseded generation can never be released again.

The one surface this design adds that the SMT did not have is account substitution — replay state now lives in a separate account rather than inside the already-validated instance. WithdrawalBitmap::validate re-derives the PDA from the instance, and both mutating instructions have a foreign-bitmap rejection test.

Not in this PR

  - Indexer/operator — ~551 references across 20 files. smt_util.rs and most of proof.rs delete outright; validate_smt_root becomes a bitmap-vs-DB check that names which nonces diverge. One behavioural trap: InvalidSmtProof was retryable and triggered a proof rebuild, whereas NonceAlreadyUsed is terminal.
  - integration/ crate (6 files) and test_utils/validator_helper.rs, which detects test-tree via a constant that disappears with smt_util.rs.
  - Docs beyond the escrow program's own — 13 files, mostly runbook term swaps.

  docs/ESCROW_PROGRAM.md, TEST_COVERAGE.md and the fuzz README are updated and verified against
  the generated IDL.
### Coverage Report

| Component | Lines Hit | Lines Total | Coverage | Artifact |
|-----------|-----------|-------------|----------|----------|
| Core | - | - | - | - |
| Indexer | - | - | - | - |
| Gateway | - | - | - | - |
| Auth | - | - | - | - |
| Withdraw Program | - | - | - | - |
| Escrow Program | - | - | - | - |
| DvP Swap Program | - | - | - | - |
| E2E Integration | - | - | - | - |
| **Total** | **-** | **-** | **-** | |

> Coverage runs in progress...

### CODE REVIEW

**Comment by sebatustra** [comment]
@greptileai

File: private-channel-escrow-program/program/src/constants.rs [hunk 1]
```diff
@@ -6,41 +6,25 @@ use pinocchio::Address;
 pub const INSTANCE_SEED: &[u8] = b"instance";
 pub const OPERATOR_SEED: &[u8] = b"operator";
 pub const ALLOWED_MINT_SEED: &[u8] = b"allowed_mint";
+pub const WITHDRAWAL_BITMAP_SEED: &[u8] = b"withdrawal_bitmap";
 
 // Instance
 pub const INSTANCE_VERSION: u8 = 1;
```
[the remark this answers is not in the document]
Comment (sebatustra) [reply private-channel-escrow-program/program/src/constants.rs:12]: the program has never been deployed on mainnet

File: README.md [hunk 2]
```diff
@@ -178,7 +178,7 @@ Within the payment channel, transactions are processed through a **five-stage pi
```
[the remark this answers is not in the document]
Comment (sebatustra) [reply README.md:178]: the commit is verified

[the remark this answers is not in the document]
Comment (sebatustra) [reply README.md:178]: yes

File: private-channel-escrow-program/TEST_COVERAGE.md [hunk 3]
```diff
@@ -6,25 +6,29 @@
...
+| Account validation paths      | 95% (19/20)  | Signer, PDA, owner, mutability, ATA program, system program   |
+| Business logic error branches | 93% (14/15)  | Nonce replay, balance verification, Token2022 extensions      |
+| Custom error codes exercised  | 100% (13/13) | All custom errors tested                                      |
+| State & trait coverage (unit) | 100% (14/14) | Instruction data parsing for all handlers                     |
+| Event coverage                | 100% (9/9)   | All events emitted in integration tests                       |
+| Security edge cases           | 100% (14/14) | Double-spend, foreign bitmap, Token2022, generation boundaries |
+| **Overall (risk-weighted)**   | **~95%**     |                                                               |
+
+> The percentages above predate the SMT-to-bitmap migration and have not been
+> re-derived against the current surface. The inventory below is accurate.
```
Comment (gitteri) [inline private-channel-escrow-program/TEST_COVERAGE.md:21]: does this mean this was generated before this code was updated?

Comment (sebatustra) [reply private-channel-escrow-program/TEST_COVERAGE.md:21]: yes, it's a hand written estimate that predates this PR. I updated the inventory below it but left the percentages alone. I'll revisit and patch the table
